### ISSUE TITLE
[Bug] CPKREDIR is unable to parse mod entries that start with square brackets

### ISSUE THREAD

**Issue by DarioSamo** [thesupersonic16/HedgeModManager#147]
Tested on 7.2-6.

Here's the output from the console:

![image](https://user-images.githubusercontent.com/538504/96497655-6acd6b00-1221-11eb-9b82-ea21e5809393.png)

Here's a sample modsDB.ini file saved from HedgeModManager that causes the issue:
```
[Main]
ReverseLoadOrder=0
ActiveMod0="[Prefix] Mod Title"
ActiveModCount=1
FavoriteModCount=0
[Mods]
[Prefix] Mod Title="C:\Program Files\Steam\steamapps\common\SonicForces\build\main\projects\exec\mods\[Prefix] Mod Title\mod.ini"
[Codes]
CodeCount=0
```
The folder name seems to be used as the key in the ini file, which causes problems if it includes square brackets due to being recognized as a new section.

Since I doubt updating CPKREDIR is an option, it seems like the most sensible solution would be to just generate a different unique key for the mod based on the folder name if possible. The above example works perfectly fine if it's replaced by something like this:
```
[Main]
ReverseLoadOrder=0
ActiveMod0="Mod Title"
ActiveModCount=1
FavoriteModCount=0
[Mods]
Mod Title="C:\Program Files\Steam\steamapps\common\SonicForces\build\main\projects\exec\mods\[Prefix] Mod Title\mod.ini"
[Codes]
CodeCount=0
```
So it's clearly just a problem with the key that is being used, given the full path works perfectly fine.

**Comment by Sajidur78** [comment]
Should be fixed with 092a068e1de701fa53fb9688fa3168a8950defbf.

**Comment by Sajidur78** [comment]
Fixed on 7.2-7.
